### PR TITLE
Update plugin auth docs

### PR DESCRIPTION
**Pull request by marcogschmidt**
Update the external auth plugin user and developer guides to reflect the recent changes in the https://github.com/solo-io/ext-auth-plugin-examples repository.

### CODE REVIEW

File: docs/content/dev/writing_auth_plugins/_index.md [hunk 1]
```diff
@@ -253,39 +256,38 @@ and to a file (`mismatched_dependencies.json`) and exit with code 1. Here's an e
...
-case we would need to add the following stanza:
+If you get an error message like this, you have to manually update the dependencies in your `go.mod` file. In this 
+case we would need to add the following entry to the `replace` section of the `go.mod` file:
 
-```toml
-[[override]]
-  name = "go.uber.org/zap"
-  version = "=v1.10.0"
+```
+go.uber.org/zap v1.13.0
```
Comment (kdorosh) [inline docs/content/dev/writing_auth_plugins/_index.md]: the full replace entry would be nice here. `x` -> `y`

Comment (marcogschmidt) [reply docs/content/dev/writing_auth_plugins/_index.md]: The output is from the actual comparison script. When the mismatch is just between 2 `require` entries, it prompts the user to just update their `require` clause.

Comment (marcogschmidt) [reply docs/content/dev/writing_auth_plugins/_index.md]: Add a link to the README file in the examples repo where I will include more info about the dependency comparison.

**Pull request merged**